src/CheckedFile.cpp

```diff
@@ -658,13 +732,20 @@ void CheckedFile::readPhysicalPage(char* page_buffer, uint64_t page)
    /// Seek to start of physical page
    seek( page*physicalPageSize, Physical );
 
+   size_t result = -1;
+   if(fd_ >= 0){
 #if defined(_MSC_VER)
-   int result = ::_read( fd_, page_buffer, physicalPageSize );

```

Comment:
Same comment as above - please don't change the return types.

`size_t` is not the same as an `int`.